src/Storages/MergeTree/MergeTask.cpp

```diff
@@ -436,7 +436,7 @@ MergeTask::StageRuntimeContextPtr MergeTask::VerticalMergeStage::getContextForNe
 bool MergeTask::ExecuteAndFinalizeHorizontalPart::execute()
 {
     assert(subtasks_iterator != subtasks.end());
-    if ((*subtasks_iterator)())
+    if ((this->**subtasks_iterator)())
```

Comment:
```
this->**subtasks_iterator
```
:cry:

Reply:
🤔

Reply:
Needs a comment about this change, otherwise, the next committer will change it to lambda with the motivation of improving the code.

Reply:
> I've added motivational comment

"Let's go code! You can do it. Achieve your goals!"